### PR TITLE
feat: integrate code plugin to RTE

### PR DESCRIPTION
**Pull request by TomasSlama**
[FX-4114]

### Description

Add support for `highlighted code like this` in RichTextEditor. The implementation is not pixel perfect with [the design](https://www.figma.com/file/q2nvjiyO2CLqBv4DeJnU3i/Product-Library-Documentation?type=design&node-id=7709-54464&mode=design&t=Ry3ddpH6WL27WQw1-0) as I think the design is not perfect. Matt has a vacation so I used common sense and merged the design with how it is working in Slack and Lexical playground. We can adjust the styles with Matt in different PR if needed.

### How to test

- [Temploy](https://picasso.toptal.net/fx-4114-rte-code)
- You can check both [RichText](https://picasso.toptal.net/fx-4114-rte-code/?path=/story/components-richtext--richtext) and [RichTextEditor](https://picasso.toptal.net/fx-4114-rte-code/?path=/story/forms-richtexteditor--richtexteditor#code-highlight) stories

### Screenshots

image

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests

**Breaking change**

- codemod is created and showcased in the changeset
- test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4114]: https://toptal-core.atlassian.net/browse/FX-4114?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/picasso-rich-text-editor/src/LexicalEditor/styles.ts [hunk 1]
```diff
@@ -160,5 +164,6 @@ export default ({ typography }: Theme) => {
         height: '22px',
       },
     },
+    code: codeStyles(theme),
```
Comment (TomasSlama) [inline packages/picasso-rich-text-editor/src/LexicalEditor/styles.ts:167]: we use the same styles for RichText component

File: packages/picasso-rich-text-editor/src/RichText/components/styles.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import type { Theme } from '@material-ui/core/styles'
+import { createStyles } from '@material-ui/core/styles'
+
+export const codeStyles = ({ palette, typography }: Theme) => ({
```
Comment (augustobmoura) [inline packages/picasso-rich-text-editor/src/RichText/components/styles.ts:4]: Just a small change, on the official [design](https://www.figma.com/file/q2nvjiyO2CLqBv4DeJnU3i/Product-Library-Documentation?type=design&node-id=7709%3A54464&mode=design&t=xcBNRS8xHR3i5xeP-1), the inline code has a small border and border-radius
image

File: packages/picasso-rich-text-editor/src/plugins/CodePlugin/CodePlugin.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+const PLUGIN_NAME = 'code'
+
+export type Props = {
+  testIds?: {
+    button?: string
+  }
+}
+
+const EmojiPlugin: RTEPlugin<Props> = ({ testIds }) => {
```
Comment (augustobmoura) [inline packages/picasso-rich-text-editor/src/plugins/CodePlugin/CodePlugin.tsx:15]: I think this is a leftover
```suggestion
const CodePlugin: RTEPlugin<Props> = ({ testIds }) => {
```

File: packages/picasso-rich-text-editor/src/RichTextEditor/story/index.jsx [hunk 4]
```diff
@@ -38,3 +38,4 @@ page
     takeScreenshot: false,
   })
   .addExample('RichTextEditor/story/ImageUpload.example.tsx', 'Image upload')
+  .addExample('RichTextEditor/story/Code.example.tsx', 'Code highlight')
```
Comment (augustobmoura) [inline packages/picasso-rich-text-editor/src/RichTextEditor/story/index.jsx:41]: I think we agreed to call it just `code` or `inline code`, while the block version is `code block`. The idea is to also add code blocks to this example?
```suggestion
  .addExample('RichTextEditor/story/Code.example.tsx', 'Inline code')
```

File: packages/picasso-rich-text-editor/src/plugins/CodePlugin/CodeButton.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import { useLexicalComposerContext } from '@lexical/react/LexicalComposerContext'
+import { Code16 } from '@toptal/picasso'
+import { $getSelection, $isRangeSelection, FORMAT_TEXT_COMMAND } from 'lexical'
+import React, { useState } from 'react'
+
+// import { TOGGLE_CODE_COMMAND } from './commands'
```
Comment (mkrl) [inline packages/picasso-rich-text-editor/src/plugins/CodePlugin/CodeButton.tsx:6]: Leftover?

File: packages/picasso-rich-text-editor/src/plugins/CodePlugin/CodePlugin.tsx (hunk 3 above)
Comment (TomasSlama) [reply packages/picasso-rich-text-editor/src/plugins/CodePlugin/CodePlugin.tsx]: done

File: packages/picasso-rich-text-editor/src/RichTextEditor/story/index.jsx (hunk 4 above)
Comment (TomasSlama) [reply packages/picasso-rich-text-editor/src/RichTextEditor/story/index.jsx]: done

File: packages/picasso-rich-text-editor/src/RichText/components/styles.ts (hunk 2 above)
Comment (TomasSlama) [reply packages/picasso-rich-text-editor/src/RichText/components/styles.ts:4]: thanks for noticing this 🙇, done ✅ 
image

File: packages/picasso-rich-text-editor/src/RichText/story/HTML.example.tsx [hunk 6]
```diff
@@ -75,7 +78,6 @@ const defaultValue: ASTType = {
           tagName: 'img',
           properties: {
             src: './jacqueline/128x88.jpg',
-            alt: 'Jacqueline',
```
[the remark this answers is not in the document]
Comment (TomasSlama) [reply packages/picasso-rich-text-editor/src/RichText/story/HTML.example.tsx:78]: oh, it was just by accident

File: packages/picasso-rich-text-editor/src/RichTextEditor/story/Code.example.tsx
[the remark this answers is not in the document]
Comment (TomasSlama) [reply packages/picasso-rich-text-editor/src/RichTextEditor/story/Code.example.tsx:1]: yeah, maybe even 1 with all plugins would be enough and one with default value 🤔

File: packages/picasso-rich-text-editor/src/plugins/CodePlugin/CodeButton.tsx (hunk 5 above)
Comment (TomasSlama) [reply packages/picasso-rich-text-editor/src/plugins/CodePlugin/CodeButton.tsx:6]: ✅

File: packages/picasso-rich-text-editor/src/RichText/story/HTML.example.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (TomasSlama) [reply packages/picasso-rich-text-editor/src/RichText/story/HTML.example.tsx:78]: ✅

File: cypress/component/RichTextEditor/CodePlugin.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+      <Container padded='small'>{value}</Container>
+    </Container>
+  )
+}
+
+const component = 'RichTextEditor'
+
+describe('CodePlugin', () => {
+  describe('when the code button in toolbar is used', () => {
+    it('inserts code tag and activate the button', () => {
```
[the remark this answers is not in the document]
Comment (TomasSlama) [reply cypress/component/RichTextEditor/CodePlugin.spec.tsx:34]: ✅

**Review by mkrl — APPROVED** [review]
Nice job! Looks good, now onto the code blocks :smile:

**Review by sashuk — APPROVED** [review]
Tested on temploy, code looks good!

**Review by augustobmoura — APPROVED** [review]
Really well structured code! GJ :+1:

**Pull request merged**